From: Alper Nebi Yasak <alpernebiyasak@gmail.com>
Date: Wed, 25 Sep 2024 10:02:46 +0200
Subject: acp: sync the mixer after selecting the port

Otherwise we might sync the wrong port and end up muted.

Fixes #4084

Origin: upstream, https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/351bc9ac3c1dec380c075a3706c81eb02f755a4a
---
 spa/plugins/alsa/acp/acp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/spa/plugins/alsa/acp/acp.c b/spa/plugins/alsa/acp/acp.c
index 353896a..bf37fff 100644
--- a/spa/plugins/alsa/acp/acp.c
+++ b/spa/plugins/alsa/acp/acp.c
@@ -1957,8 +1957,8 @@ int acp_device_set_port(struct acp_device *dev, uint32_t port_index, uint32_t fl
 		d->mixer_path = data->path;
 		mixer_volume_init(impl, d);
 
-		sync_mixer(d, p);
 		res = pa_alsa_ucm_set_port(d->ucm_context, p);
+		sync_mixer(d, p);
 	} else {
 		pa_alsa_port_data *data;
 
